Exercise2/Plotter.py

```diff
@@ -0,0 +1,39 @@
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+
+#Creates class
+class Plot(object):
+    def __init__(self):
+        self.t = np.linspace(-10,10,1000)
+        self.y = np.linspace(-10,10,1000)
+        self.i = 0
+
+    def Lambda(self, t):
+        return 5 * math.sin(2 * t * math.pi)
+
+    def h(self, t):
+        return 3*math.pi**(0-self.Lambda(t))
+
+    def execute(self):
+        for item in self.t:
+            self.y[self.i] = self.h(self.t[self.i])
+            self.i +=1
```

Comment:
Here I have two suggestions:

Instead of using `self.i` and keeping track of this variable, you could define an ind variable which will iterate for the length of `self.t`. Like this:

```
for ind, t_value in enumerate(self.t):
    self.y[ind] = self.h(t_value)
```

Another suggestion could be to work with numpy. Let's imagine that you redefine your `_lambda` and `h` methods to work with the `self.t` as a numpy array directly. Then you could simply do:

```suggestion
        self.y = self.h(self.t)
```

And this would avoid even having the for loop. It's an advanced option, but I'm sure that you will have no troubles learning about it after some time working with us!